nasa/harmony
# HARMONY-1654: Refactor code and update indices created on the jobs table

## Jira Issue ID
HARMONY-1654

## Description
The ticket was to fix a case where we were pulling back job links from one of the workflow-ui queries. When looking into it I found that our internal code APIs for querying for jobs were inconsistent and made it easy to pull back job links unintentionally. I refactored and tried to use common code for querying for all jobs or querying for a single job. By default links are not returned, the caller explicitly needs to request links now.

I also fixed all of the indices on the jobs table to match the queries that we execute.

## Local Test Steps
1. Run the database migration and verify it succeeds.
2. Regression test the jobs listing and jobs status page. Verify listings do not include the job links and the job status does include job links. For large requests make sure the paging works correctly for the jobs status page.
3. Make sure all pages on the workflow-ui work as expected.
4. Verify harmony-py works as before when submitting requests and downloading results.

I tested out with a sandbox run using locust where I ran the migrations with 150 concurrent requests. I created over 22K jobs and tested with large jobs with 247K granules. The migrations did run into deadlocks a couple times, but they did succeed with retries on a single deployment (we retry up to 10 times). Cloudwatch did not show any slow queries and the original slow queries that were logged that were the reason for this ticket did not appear.

## PR Acceptance Checklist
